src/content-handlers/iiif/modules/uv-openseadragoncenterpanel-module/OpenSeadragonCenterPanel.ts

```diff
@@ -932,6 +931,10 @@ export class OpenSeadragonCenterPanel extends CenterPanel<
         this.viewer.addOverlay(div, rect);
       }
     }
+
+    if (annotations.length && this.shouldZoomToInitialAnnotation()) {
+      this.zoomToInitialAnnotation();
```

Comment:
Previously, this was happening in response to the `IIIFEvents.ANNOTATIONS` event, but now it's presumably happening just once. I'm not familiar with when/how this event is triggered, but is it possible this could be a problem? Is there any reason not to move this check into the event handler instead of leaving it here in the function?

Reply:
Hi @demiankatz, I think it should be fine, as I've moved it into overlayAnnotations() which is called in the event handler, so it amounts to the same thing as far as I can tell.

At the time I was thinking two things:

1) It allows checking annotations.length and therefore not running the zoom code if there aren't any. I'm not sure if IIIFEvents.ANNOTATIONS would fire if that were the case, but I wanted to be safe rather than go digging through the code. Though I may do so now just to satisfy my curiosity :)

2) overlayAnnotations() is also called from within openPagesHandler() and I figured that any time there were annotations being drawn we should be checking for the zoom setting and zoom if true.

Reply:
I figured you had probably given this some thought; thanks for sharing the explanation. It makes sense to me (within the context of my own fairly limited knowledge). :-)

Things do seem to work as expected with regard to the automatic Zoom. However, while testing this, I ran into a couple of other things that are broken. I strongly suspect that these problems are not directly related to the changes here, but I'm seeing significantly different behavior at https://universalviewer.dev (which I thought was supposed to track the dev branch, but which actually seems to be frozen in time a little further back). We probably need to open separate issues/PRs to address these things, but I'm first going to mention them here in case you're seeing different behavior on your end, or have any ideas about where the problems came from:

Issue 1: there's a z-index problem, so that if I open the settings pop-up while viewing a highlighted page, the highlight boxes float over the settings. This is a problem in this PR, but not in universalviewer.dev. I also notice that the highlight boxes seem to be a slightly different color, so there must have been a styling change somewhere along the line.

Issue 2: the markers showing search results along the page track at the bottom of the viewer don't work consistently. I'm seeing slightly different behavior here and in universalviewer.dev, but in both instances, some markers will jump to a page when clicked and others will not. Sometimes clicking one works once and does not work again later. For example, in the example "phillips" search cited here, I can jump to the first hit, and then I can jump to the hit on page 165, but most of the other hits are non-functional.

I'm not sure how best to proceed. It's _probably_ okay to merge this as incremental progress, but I would want to do some proper A/B testing first to be sure those other issues aren't related (which I can't do quickly with universalviewer.dev right now due to it being out of sync, but which I can do with a local environment later when I have a bit more time). I suspect that issue 2 is a whole separate thing that needs to be addressed, and maybe if we tackle that first it will make it a little easier to test/exercise this one. And issue 1 is just a minor nuisance, but maybe worth addressing too.

Any thoughts on how you'd like to move forward from here?

Reply:
Re 1, in the openseadragon center module's style.less file, the selector `.annotationRect` seems to have gotten a z-index of 20

Looks like it was in this commit: https://github.com/universalviewer/universalviewer/commit/1f768803803f966c1cfb4b25c9360b4276ebcbeb

in this PR: https://github.com/UniversalViewer/universalviewer/pull/1052

Either that'll need removing, or the dialogue will need a higher one. I'll try to fix it quickly in this PR and see which works best.

Re 2, The issue seems to be that only when the main body of the tooltip is clicked on will the page change.

Clicking the marker doesn't seem to work, and depending on where you click it will sometimes trigger the tooltip to close. I think this definitely needs to be a new issue.

Reply:
@demiankatz I've committed a z-index change to the overlays container (from 10 to 30) that seems to fix the issue. 

I've chosen a value that should keep it above the annotations, but remains lower than other pre-existing z-indexes which are between 100 and 1000, so any other behaviours should remain the same.